### PR TITLE
Refactor TeamsActivityHandler with refactored ActivityHandler class

### PR DESCRIPTION
**Pull request by stevengum**
Fixes #

## Specific Changes

  - 
  -
  -

### CODE REVIEW

File: libraries/botbuilder/src/teamsActivityHandler.ts [hunk 1]
```diff
@@ -9,29 +9,119 @@ import {
...
+     * @param context 
      */
-    public onInvokeActivity(handler: (context: TurnContext) => Promise<void>): this {
-        this.checkRegisteredTeamsHandlers('InvokeActivity');
-        return this.on('InvokeActivity', async (context, next) => {
-            await handler(context);
-        });
+    protected async onTurnActivity(context: TurnContext): Promise<void> {
+        if (context.activity.channelId === 'msteams') {
+            // this.teamsRosterClient =
```
Comment (daveta) [inline libraries/botbuilder/src/teamsActivityHandler.ts]: >this.teamsRosterClient [](start = 15, length = 22)

Hmm.

File: libraries/botbuilder/src/teamsActivityHandler.ts [hunk 2]
```diff
@@ -9,29 +9,119 @@ import {
...
+                    case 'task/fetch':
+                        const fetchResponse = await this.onTeamsTaskModuleFetch(context, context.activity.value);
+                        return TeamsActivityHandler.createInvokeResponse(fetchResponse);
+
+                    case 'task/submit':
+                        const submitResponse = await this.onTeamsTaskModuleSubmit(context, context.activity.value);
+                        return TeamsActivityHandler.createInvokeResponse(submitResponse);
+
+                    default:
+                        throw new Error('NotImplemented');
```
Comment (daveta) [inline libraries/botbuilder/src/teamsActivityHandler.ts]: >NotImplemented [](start = 41, length = 14)

Was this going to be a 400 or 404?

Comment (stevengum) [reply libraries/botbuilder/src/teamsActivityHandler.ts]: I just pushed another commit that changes this to 400

File: libraries/botbuilder/src/teamsActivityHandler.ts (hunk 1 above)
Comment (stevengum) [reply libraries/botbuilder/src/teamsActivityHandler.ts]: No impl yet on the class for this member, leaving as is

**Pull request merged**